### PR TITLE
Analytics Package Integration

### PR DESCRIPTION
**Pull request by jeremy6462**
This PR integrates the new analytics events package into the data analyzer. It replaces all instances of event code constants with enum raw values stored in the swift package. Along the way, I also updated some queries to use the base count query function.

### CODE REVIEW

File: AppBookDataAnalyzer/Analytics for Rows/Career Explorer/JobFavoritedCount.swift [hunk 1]
```diff
@@ -65,8 +64,8 @@ private struct JobFavoriteStatus {
                 FROM \(Database.EventLog.tableName)
                 WHERE \(Database.EventLog.Column.appbookId) = \(appbook.id)
                 AND \(Database.EventLog.Column.pageNumber) = \(pageNumber)
-                AND \(Database.EventLog.Column.code) = \(jobFavoritedEventCode)
-                AND \(Database.EventLog.Column.description) = '\(favoritedEventPrefix) \(isFavorited)'
+                AND \(Database.EventLog.Column.code) = \(AppBookAnalyticEvent.jobFavorited.code)
+                AND \(Database.EventLog.Column.description) = '\(favoritedEventPrefix) - \(isFavorited)'
```
Comment (jeremy6462) [inline AppBookDataAnalyzer/Analytics for Rows/Career Explorer/JobFavoritedCount.swift:68]: This was missing from the analytic and would have resulted in a 0 count

File: AppBookDataAnalyzer/Database/Database+BaseEventCount.swift [hunk 2]
```diff
@@ -16,13 +17,15 @@ extension Database {
     ///   - appbookId: The `AppBook.ID`, most likely present in the `TextbookMaterial` passed in during the `anaylze` function for an `Analytic`
```
Comment (fjfur10) [inline AppBookDataAnalyzer/Database/Database+BaseEventCount.swift:17]: Just a nitpick, we should update the documentation to reflect the changes with the new package (eventCodes-> events)

**Review by fjfur10 — APPROVED** [review]
This looks very neat and organized! I just have a  very small nitpick for documentation

File: AppBookDataAnalyzer/Database/Database+BaseEventCount.swift (hunk 2 above)
Comment (jeremy6462) [reply AppBookDataAnalyzer/Database/Database+BaseEventCount.swift:17]: good call!

**Pull request merged**